src/k8s/pkg/k8sd/setup/kube_apiserver.go

```diff
@@ -79,8 +79,13 @@ func KubeAPIServer(snap snap.Snap, serviceCIDR string, authWebhookURL string, en
 	switch datastore {
 	case "k8s-dqlite":
 		args["--etcd-servers"] = fmt.Sprintf("unix://%s", path.Join(snap.K8sDqliteStateDir(), "k8s-dqlite.sock"))
+	case "external":
+		args["--etcd-servers"] = datastoreUrl
+		args["--etcd-cafile"] = path.Join(snap.EtcdPKIDir(), "ca.crt")
+		args["--etcd-certfile"] = path.Join(snap.EtcdPKIDir(), "client.crt")
+		args["--etcd-keyfile"] = path.Join(snap.EtcdPKIDir(), "client.key")
```

Comment:
we still need some way to ensure these are only set if TLS certs have in fact been specified, right? for example, in the controller-manager, we do https://github.com/canonical/k8s-snap/blob/main/src/k8s/pkg/k8sd/setup/kube_controller_manager.go#L25-L29